scripts/west_commands/runners/bossac.py

```diff
@@ -91,7 +91,7 @@ def is_extended_samba_protocol(self):
     def is_partition_enabled(self):
         build_conf = BuildConfiguration(self.cfg.build_dir)
 
-        if 'CONFIG_USE_DT_CODE_PARTITION' not in build_conf:
+        if build_conf.get('CONFIG_USE_DT_CODE_PARTITION', 'n') != 'y':
```

Comment:
This should also work:
```suggestion
        if build_conf.get('CONFIG_USE_DT_CODE_PARTITION') != 'y':
```